schemas/user.schema.ts

```diff
@@ -0,0 +1,33 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document, Types } from 'mongoose';
+
+export type UserDocument = User & Document<Types.ObjectId>;
+
+export type UserDocumentWithoutPassword = Omit<User, 'password'> &
+  Document<Types.ObjectId>;
+
+@Schema()
+export class User {
+  @Prop({ required: true })
+  first_name: string;
+
+  @Prop({ required: true })
+  last_name: string;
+
+  @Prop({ required: true, select: false })
+  password: string;
```

Comment:
Are you sure its a good practice to store password directly in db? It is not secure, you should use password hashing https://docs.nestjs.com/security/encryption-and-hashing

Reply:
This is just how I named this field. I hash the password on creation and update
<img width="488" alt="Screenshot 2024-08-18 at 18 07 14" src="https://github.com/user-attachments/assets/a1dfcd93-1129-41f0-8fff-ea194f8664ae">

Reply:
Sorry, missed this. Actually, maybes its better to move hashing logic to db layer? If you use repository in another place, your password will be stored without hashing

Reply:
That's a great idea